workflows/pipe-common/pipeline/hpc/avail.py

```diff
@@ -0,0 +1,58 @@
+#  Copyright 2017-2023 EPAM Systems, Inc. (https://www.epam.com/)
```

Comment:
I think it is better to emphasize that it is about instance, so how about to name this file as `instance_avail`

Reply:
I've extracted `instance` package with `avail`, `provider` and `selector` subpackages.

Reply:
LGTM